reference/crosswalk-aws/vpc.md

```diff
@@ -0,0 +1,499 @@
+components, including ECS and EKS clusters, API gateways, and various network load balancing options.
+
+Each account has a default regional network and VPC to make it easy to get up and running. Most production
+circumstances call for dedicated VPCs and network isolation. This includes multi-tenanted scenarios where VPCs can be
+used for strong network isolation between endpoints and resources that are otherwise sharing an AWS account.
+
+Pulumi Crosswalk for AWS provides simple, out of the box VPC functionality that follows widely accepted best
+practices. This ensures you can provision and evolve your VPCs across many environments productively and safely,
+without needing to recreate the same VPC templates for every new project you tackle.
+
+## Managing VPCs
+
+The [aws.ec2.Vpc](https://pulumi.io/reference/pkg/nodejs/@pulumi/aws/ec2/#Vpc) class provides full access to the
+AWS VPC API, and [aws.ec2](https://pulumi.io/reference/pkg/nodejs/@pulumi/aws/ec2) the entire AWS EC2 API. Using
+these packages, you can configure all aspects of AWS networks for your applications and infrastructure.
+
+The [awsx.ec2.Vpc](https://pulumi.io/reference/pkg/nodejs/@pulumi/awsx/ec2/#Vpc) class encapsulates a complete
+configuration of an AWS network, including the actual VPC itself, in addition to public and/or private subnets, route
+tables, and gateways, across multiple availability zones. It is designed to be easier to use, with reasonable defaults,
+and follows AWS's own best practices, with configurability for advanced scenarios. The two can be used together.
+
+Below are some of the most common infrastructure as code tasks with VPCs.
+
+## Getting the Default VPC
+
+Often resources like clusters, API gateways, lambdas, and more, will request a VPC object or ID. This ensures
+such resources inside of your VPC so network traffic are isolated from other VPCs in your account.
+
+So where do you get such a VPC from? One way is that each AWS account has a default VPC per region. Using the default
+VPC is often the easiest path when you're just getting up and running or don't yet understand your specific networking
+requirements. Most resources will use this default VPC automatically if you leave it unspecified. In other cases,
+you may be required to pass it explicitly, in which case you'll need to get it programmatically.
+
+To get the default VPC, just call the `awsx.ec2.Vpc.getDefault()` function:
+
+```typescript
+import * as awsx from "@pulumi/awsx";
+
+// Fetch the default VPC information from your AWS account:
+const vpc = awsx.ec2.Vpc.getDefault();
+
+// Export a few interesting fields to make them easy to use:
+export const vpcId = vpc.id;
+export const vpcPrivateSubnetIds = vpc.privateSubnetIds;
+export const vpcPublicSubnetIds = vpc.publicSubnetIds;
+```
+
+This example reads the default VPC and exports some of its properties for easy consumption. `pulumi up` will show:
+
+```bash
```

Comment:
Despite downgrading to an earlier version of Hugo (v0.55.4) to workaround some markdown rendering regressions, I still found one case that Hugo was having trouble rendering on v0.55.4. For some reason, specifying a language for code fences that contain the output of `pulumi up` was causing the code fence not to render as a code snippet.

I had to workaround this by remove the wrapping ```bash ... ``` and indenting the code snippet twice (e.g. 8 spaces) for it to render correctly as a code snippet (just without a language for highlighting, which is reasonable to live without in this case).

I need to track down whether there is already a bug open on this or if I need to open one against [Hugo's default markdown renderer](https://github.com/russross/blackfriday).

It *might* not be an issue here, as the places I was running into this was in our tutorials where the code snippets are inside lists of steps, and haven't narrowed it to a minimal repro. So just an FYI if this doesn't render as you expect, try removing the wrapping ```bash ... ``` and indenting twice instead.